google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerImpl.java

```diff
@@ -67,6 +81,26 @@ final class TransferManagerImpl implements TransferManager {
                       ImmutableMap.of("User-Agent", USER_AGENT_ENTRY + LIBRARY_VERSION)))
               .build();
     }
+    // Create the blobWriteSessionConfig for ParallelCompositeUpload
+    ParallelCompositeUploadBlobWriteSessionConfig pcuConfig =
+        BlobWriteSessionConfigs.parallelCompositeUpload()
+            .withExecutorSupplier(ExecutorSupplier.useExecutor(executor));
+    if (storageOptions instanceof GrpcStorageOptions
+        && transferManagerConfig.isAllowParallelCompositeUpload()) {
+      storageOptions =
+          ((GrpcStorageOptions) storageOptions)
+              .toBuilder()
+              .setBlobWriteSessionConfig(pcuConfig)
+              .build();
+    } else if (storageOptions instanceof HttpStorageOptions
+        && transferManagerConfig.isAllowParallelCompositeUpload()) {
+      storageOptions =
+          ((HttpStorageOptions) storageOptions)
+              .toBuilder()
+              .setBlobWriteSessionConfig(pcuConfig)
+              .build();
+    }
```

Comment:
If you rebase against `main` this isn't necessary any longer, `setBlobWriteSessionConfig` has been added to `StorageOptions`

```suggestion
    // Create the blobWriteSessionConfig for ParallelCompositeUpload
    if (transferManagerConfig.isAllowParallelCompositeUpload()) {
      ParallelCompositeUploadBlobWriteSessionConfig pcuConfig =
          BlobWriteSessionConfigs.parallelCompositeUpload()
              .withExecutorSupplier(ExecutorSupplier.useExecutor(executor));
      storageOptions =
          storageOptions
              .toBuilder()
              .setBlobWriteSessionConfig(pcuConfig)
              .build();
    }
```